File: OldSolvedProblems/20ValidParentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> tower;
        
        for(int i = 0; i < s.length(); i++) {
            if(s[i] == '(' || s[i] == '[' || s[i] == '{') {
                tower.push(s[i]);
            } else if(!tower.empty() && (s[i]==')'&&tower.top()=='(' || s[i]==']'&&tower.top()=='[' || s[i] =='}'&&tower.top() == '{')) {
                    tower.pop();
            } else {
                return false;
            }
        }
        if(tower.empty()) {
            return true;
        }
        return false;
    }
};

File: Stacks/20ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        # Python stacks suck
        def peak(stack):
            if not stack:
                return None
            return stack[-1]
        
        validStart = ["(", "[", "{"]
        stack = []
        for i in range(len(s)):

            if s[i] in validStart:
                stack.append(s[i])
            else:
                if s[i] == ")" and peak(stack) == "(" or s[i] == "]" and peak(stack) == "[" or s[i] == "}" and peak(stack) == "{":
                    stack.pop()
                else:
                    return False
        
        if len(stack) == 0:
            return True
        return False